setup.py

```diff
@@ -101,7 +101,7 @@ def recursive_data_files(directory, install_directory):
     "craft-archives",
     "craft-cli",
     "craft-grammar",
-    "craft-parts",
+    "craft-parts>=1.30.1",
```

Comment:
After https://github.com/canonical/snapcraft/issues/4843, I discussed with @lengau and agreed that we should set minimum versions in `setup.py`.